### ISSUE TITLE
Pipewire and changing the audio stream causes the video to freeze

### ISSUE THREAD

**Issue by ImAReplicant** [xbmc/xbmc#20996]
### Describe the bug
Here is a clear and concise description of what the problem is:

On Archlinux with pipewire-pulse package : 
When I change the audio stream of a video, the video freezes for ~5sec and I have no sound. The sound comes back ~30sec later but it stutters. I need to stop and replay the video to stop the stuttering.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

No freezing when changing audio streams

## Actual Behavior

The video freezes and the audio stutters when changing audio streams.

## Possible Fix

Replacing pipewire-pulse with pulseaudio.

### To Reproduce

1. Install pipewire-pulse
2. Play a video
3. Change the audio stream of the video

### Debuglog

The debuglog can be found here:

[kodi.log](https://paste.kodi.tv/maxalotuso.kodi)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Archlinux
 - Kodi version: 19.3-2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by fritsch** [comment]
Does this mean: if you remove pipewire-pulse it works nicely? - from the logs the kodi pulseaudio sink is used and nothing about pipewire appears. Is the system below that "transparently" pipewiring that to somewhere else?

**Comment by ImAReplicant** [comment]
Yes, on archlinux pipewire-pulse and pulseaudio are in conflict, so install pulseaudio removes pipewire-pulse.
[https://archlinux.org/packages/extra/x86_64/pipewire-pulse/](https://archlinux.org/packages/extra/x86_64/pipewire-pulse/)

Sorry, I don't quite understand the last question. 
If this is a technical question about pipewire, I don't know how pipewire-pulse works, sorry.

**Comment by fritsch** [comment]
Okay - you are using 19.3 which does not yet have a "native pipewire" sink. I see two options:

a) You use the real pulseaudio which our implementation supports "natively". In that case you have to file a bug with ARCH as it seems like the pipewire-pulse does something which is incompatible with pulseaudio users that actually care about "latency", which also from the log seems the culprit. The get_latency call just returns bogs.

b) You keep pipewire-pulse installed and try to use kodi version 20-alpha instead. Here we have native pipewire audio sink support and are happy for testers.

On the forums (without the debuglog) I assumed you were having issues with our pipewire sink, which is not the case - as said in a) sadly I cannot fix the pipewire-pulse compatibility layer as kodi's pulse implementation is verified by the pulseaudio devs and working standard compliant I don't want to introduce "guesswork" upon an invisible pipewire transport below.

**Comment by ImAReplicant** [comment]
Ok I will use pulseaudio for now, it's not a problem. 
I'll file a bug report with pipewire because it's not only arch I also have this problem with fedora which uses pipewire by default.
